src/types/header.ts

```diff
@@ -45,8 +56,18 @@ const COINBASE = "0xabde1".padEnd(42, "0");
  * internal type.
  */
 export function toEthHeader(
-  { header, gasUsed, logsBloom, receiptRoot, transactionRoot }: {
+  {
+    header,
+    blockNumber,
+    blockHash,
```

Comment:
where do we use those?

Reply:
so I need to convert the blockNumber to hex and pad it, same for the blockHash (need to pad it) so instead of doing it in every function, I just do it once in the main and then pass it to every function.